lib/bq_stream/bq_stream.rb

```diff
@@ -44,16 +42,16 @@ def create_bq_writer
     end
 
     def config_initialized
-      QueuedItem.build_table
-      OldestRecord.build_table
       create_bq_writer
       create_bq_dataset unless @bq_writer.datasets_formatted.include?(dataset)
       create_bq_table unless @bq_writer.tables_formatted.include?(bq_table_name)
       initialize_old_records if back_date
     end
 
     def initialize_old_records
-      log(:info, "#{Time.now}: $$$$$$$$$$ Start Init Old Records $$$$$$$$$$")
+      OldestRecord.delete_all
+      OldestRecord.create(table_name: '! revision !', attr: `cat #{File.expand_path ''}/REVISION`)
```

Comment:
So any time you restart catprint it is going to remove all oldest records?

Reply:
Yes, it will then reinitialize the latest date for each of the bq_attributes. This is since rows are deleted as we reach the back date and if there are any additions or deletions to any bq_attributes in the system the table will need to reflect that.